### PR TITLE
FCN-123: additional security groups

### PR DESCRIPTION
**Pull request by davidaznaur**
## Description
A new expandable section to display additional security groups after user selects a VPC in the machine poolls step.
If there are no additional security groups present, then the user will see an alert indicating that no additional security groups exist with a link to aws console and a "button" (link) to refresh additional security groups. 
If additional security groups exist, user will be presented with an alert saying that user wont be able to add or edit security groups associated with machine pools that were created during cluster creation, and a multiselect dropdown (custom component)

**Jira issue #** 
[FCN-123](https://issues.redhat.com/browse/FCN-123)

**Backport of** #

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature/enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] UI/UX improvement
- [ ] Code refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Tests
- [ ] Configuration change
- [ ] Infrastructure/build change
- [ ] Other (please specify):

### Manual Testing

**Test Steps:**
1. Launch storybook and navigate to machine pools step
2. Select a VPC (test-vpc-1 has security groups, test-vpc-2 does not)
3. Scroll to the bottom and expand Advanced machine pool configuration. Below Root disk size there is another expandable section with additional security groups.
4. If you selected test-vpc-1, select several additional security groups. Then proceed and select different vpc. Confirm that additional security groups are cleared when user selects different VPC

**Test Environment:**
- [X] Tested locally
- [X] Tested in Storybook

### Automated Testing

**E2E Run:**

- [ ] Cypress Tests: E2E tests completed successfully (npm run e2e:run)
      

- [ ] Playwright Tests: E2E tests completed successfully (cd playwright && npm run live)
      

**Unit Tests:**
- [ ] Unit tests added/updated
- [ ] All unit tests pass (npm run test)

**Integration Tests:**
- [ ] Integration tests added/updated
- [ ] All integration tests pass

### After
image
image
image

### Code Quality
- [ ] My code follows the project's style guidelines
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have removed any console logs and debugging code
- [ ] My changes generate no new warnings or errors
- [ ] I have checked for and fixed any linting errors (npm run lint)
- [ ] Code formatting is correct (npm run prettier:fix)

### Documentation
- [ ] I have updated the documentation accordingly
- [ ] I have updated the README if necessary
- [ ] I have added/updated Storybook stories for new/modified components
- [ ] I have updated TypeScript types/interfaces

### Accessibility
- [ ] My changes follow accessibility best practices
- [ ] Interactive elements are keyboard accessible
- [ ] Proper ARIA labels and roles are used where needed
- [ ] Color contrast meets WCAG standards

### Dependencies
- [ ] Any dependent changes have been merged and published
- [ ] I have updated package dependencies if needed

## Breaking Changes

**Does this PR introduce breaking changes?**
- [ ] Yes
- [ ] No

### Focus Areas

- 

### Questions for Reviewers

- 

---

## Summary by Sourcery

Add an expandable security groups configuration section to the ROSA machine pools wizard, including supporting types, helpers, and storybook coverage.

New Features:
- Introduce an Additional security groups expandable section on the machine pools step that allows selecting worker security groups per VPC.
- Show contextual alerts when security groups cannot be edited for existing machine pools and when a VPC has no associated security groups, with links to AWS and ROSA docs.

Enhancements:
- Extend VPC and wizard helper types to include security groups metadata and truncation utilities for long names.
- Add a reusable form helper-text component to standardize validation and error display for form fields.

Tests:
- Add Playwright component tests and Storybook stories to cover security groups selection behavior, empty states, and VPC change handling.

### CODE REVIEW

**Review by jloss-redhat — DISMISSED** [review]

**Comment by davidaznaur** [comment]
@jloss-redhat and @nitin-dhevar had to rearrange fields a little bit, based on the input from Margot

image

File: src/components/Wizards/types.ts [hunk 1]
```diff
@@ -50,8 +50,15 @@ export type CIDRSubnet = {
   subnet_id: string;
 };
 
+export type SecurityGroup = {
```
Comment (nitin-dhevar) [inline src/components/Wizards/types.ts:60]: @davidaznaur We have duplicate SecurityGroup type defined in both /SecurityGroupSection/types.ts and /Wizards/types.ts‎, shall we remove the types file in SecurityGroupSection dir and move all of the definitions to common Wizards file

**Review by nitin-dhevar — CHANGES_REQUESTED** [review]

**Review by jloss-redhat — DISMISSED** [review]

**Review by nitin-dhevar — APPROVED** [review]

**Review by jloss-redhat — APPROVED** [review]
